docker-app/qfieldcloud/core/serializers.py

```diff
@@ -535,3 +539,44 @@ class LatestPackageSerializer(serializers.Serializer):
     package_id = serializers.UUIDField()
     packaged_at = serializers.DateTimeField()
     data_last_updated_at = serializers.DateTimeField()
+
+
+class TeamSerializer(serializers.ModelSerializer):
+    organization = serializers.StringRelatedField(source="team_organization")
+
+    class Meta:
+        model = Team
+        fields = ("username", "organization")
+
+
+class TeamMemberSerializer(serializers.ModelSerializer):
+    member = serializers.CharField()
+
+    def to_internal_value(self, data):
+        validated_data = super().to_internal_value(data)
+        email_or_username = data.get("member")
+        existing_username = self.context["view"].kwargs.get("member_username")
+
+        if existing_username is not None:
```

Comment:
Managed to check if the request method is post or put/patch by checking the kwargs and not by checking the type of request

Reply:
I guess this comment related to the line below. In any case, you should not go to such extent to check for the kwargs.

The PATCH and PUT on already existing team members is not needed. One should just delete the old one and add a new member.